crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -145,4 +147,101 @@ impl WellKnownNote {
             },
         }
     }
+
+    /// Checks the correctness of the provided note inputs against the target account.
+    ///
+    /// It performs:
+    /// - for all notes: a check that note inputs have correct number of values.
+    /// - for `P2ID` note: assertion that the account ID provided by the note inputs is equal to the
+    ///   target account ID.
+    /// - for `P2IDR` note: assertion that the account ID provided by the note inputs is equal to
+    ///   the target account ID (which means that the note is going to be consumed by the target
+    ///   account) or that the target account ID is equal to the sender account ID (which means that
+    ///   the note is going to be consumed by the sender account)
+    pub fn check_note_inputs(
+        &self,
+        note: &Note,
+        target_account_id: AccountId,
+        block_ref: BlockNumber,
+    ) -> NoteAccountCompatibility {
+        match self {
+            WellKnownNote::P2ID => {
+                let note_inputs = note.inputs().values();
+                if note_inputs.len() != 2 {
```

Comment:
nit: instead of hard-coding `2` here (and other values in other places), we should probably create a constant for it. We could also add `WellKnownNote::num_expected_inputs()` and then this line becomes:
```rust
if note_inputs.len() != self.num_expected_inputs() {
```